pennylane/math/utils.py

```diff
@@ -241,13 +244,23 @@ def requires_grad(tensor):
 
         return should_record_backprop([tf.convert_to_tensor(tensor)])
 
-    if interface in ("torch", "autograd"):
-        return tensor.requires_grad
+    if interface == "autograd":
+        if isinstance(tensor, ArrayBox):
+            return True
+
+        # Currently, in the Autograd interface, we assume
+        # that all objects are differentiable by default.
+        return getattr(tensor, "requires_grad", True)
+
+    if interface == "torch":
+        return getattr(tensor, "requires_grad", False)
 
     if interface == "numpy":
         return False
 
     if interface == "jax":
-        return True
+        import jax
+
+        return isinstance(tensor, jax.interpreters.ad.JVPTracer)

```

Comment:
nice! Is this a hack or intended way?

Reply:
Not so much a hack as 'undocumented behaviour'. Since it is not a public API, JAX could easily change it without warning in a future version.